models/intermediate/int_zendesk__schedule_spine.sql

```diff
@@ -90,21 +95,226 @@ with timezone as (
+        {{ dbt.datediff("holiday_week_start", "holiday_start_date_at", "minute") }} as minutes_from_sunday_start,
+        {{ dbt.datediff("holiday_week_start", "holiday_end_date_at", "minute") }} as minutes_from_sunday_end
+    from holiday_start_end_times
+
+-- Determine which schedule days include a holiday
+), holiday_check as (
+
+    select
+        *,
+        case when minutes_from_sunday_start < start_time_utc and minutes_from_sunday_end > end_time_utc 
+            then holiday_name 
+        end as holiday_name_check
+    from holiday_minutes
+
+-- Consolidate the holiday records that were just created
+), holiday_consolidated as (
+
+    select 
+        schedule_id, 
+        time_zone, 
+        schedule_name, 
+        valid_from, 
+        valid_until, 
+        start_time_utc, 
+        end_time_utc, 
+        holiday_week_start,
+        cast({{ dbt.dateadd("second", "86399", "holiday_week_end") }} as {{ dbt.type_timestamp() }}) as holiday_week_end,
+        max(holiday_name_check) as holiday_name_check
+    from holiday_check
+    {{ dbt_utils.group_by(n=9) }}
+
+-- Since we have holiday schedules and normal schedules, we need to union them into a holistic schedule spine
+), spine_union as (
+
+    select
+        schedule_id, 
+        time_zone, 
+        schedule_name, 
+        valid_from, 
+        valid_until, 
+        start_time_utc, 
+        end_time_utc, 
+        holiday_week_start,
+        holiday_week_end,
+        holiday_name_check
+    from holiday_consolidated
+
+    union all
+
+    select
+        schedule_id, 
+        time_zone, 
+        schedule_name, 
+        valid_from, 
+        valid_until, 
+        start_time_utc, 
+        end_time_utc, 
+        null as holiday_week_start,
+        null as holiday_week_end,
+        null as holiday_name_check
     from calculate_schedules
+
+-- Within the spine, determine which schedules contain a holiday
+), valid_adjustment as (
+
+    select 
+        *,
+        case when max(holiday_name_check) over (partition by schedule_id, holiday_week_start order by holiday_week_start rows unbounded preceding) is not null then 1 else 0 end as is_holiday_week
+    from spine_union
+
+-- Now that we have an understanding of which weeks are holiday's let's consolidate them with non holiday weeks
+), all_periods as (
+
+    select distinct
+        schedule_id,
+        holiday_week_start as period_start,
+        holiday_week_end as period_end,
+        start_time_utc,
+        end_time_utc,
+        holiday_name_check,
+        true as is_holiday_week
+    from valid_adjustment
+    where holiday_week_start is not null
+        and holiday_week_end is not null
+
+    union all
+
+    select distinct
+        schedule_id,
+        valid_from as period_start,
+        valid_until as period_end,
+        start_time_utc,
+        end_time_utc,
+        cast(null as {{ dbt.type_string() }}) as holiday_name_check,
+        false as is_holiday_week
+    from valid_adjustment
```

Comment:
For the non-holiday week portion of this union, do we need to add the contrary of line 207-208?

Reply:
Hmmm that is a good call out. I will need to confirm this a bit more. I do recall it was needed to get the schedule to properly generate. However, the data access has since been removed so we can't investigate this further at the moment. Let me do some more digging here.